js/product-details.js

```diff
@@ -0,0 +1,155 @@
+  var containerColumn = document.getElementsByClassName("product-details")[0];
+  var containerRow = document.getElementsByClassName(
+    "product-details-value"
+  )[0];
+  if (typeof drink[key] == "object" && Array.isArray(drink[key]) != true) {
+    containerIfValueObject(drink[key], containerColumn, containerRow)
+  } else if (Array.isArray(drink[key])) {
+    containerIfValueArray(drink[key], containerColumn, containerRow)
+  } else {
+    containerIfValueLine(drink[key], containerColumn, containerRow);
+  }
+};
+
+const containerIfValueObject = function (value, containerColumn, containerRow) {
+  let drinkKey = value;
+  let valuecontainerColumn = document.createElement("div");
+  let valuecontainerRow = document.createElement("div");
+  for (key2 in drinkKey) {
+    if (drinkKey.hasOwnProperty(key2)) {
+      buildValueKey(key2, valuecontainerColumn)
+      buildValueKey(key2, valuecontainerRow)
+      if (Array.isArray(drinkKey[key2])) {
+        forrEachArrayValue(drinkKey[key2], valuecontainerColumn, valuecontainerRow);
+      } else {
+        buildArrayValue(drinkKey[key2], valuecontainerColumn)
+        buildArrayValue(drinkKey[key2], valuecontainerRow)
+      }
+    }
+  }
+  containerColumn.appendChild(containerAddClass(valuecontainerColumn, "column"));
+  containerRow.appendChild(containerAddClass(valuecontainerRow, "row"));
+}
+
+const containerIfValueArray = function (a, containerColumn, containerRow) {
+  let valuecontainerColumn = document.createElement("div");
+  let valuecontainerRow = document.createElement("div");
+  forrEachArrayValue(a, valuecontainerColumn, valuecontainerRow);
+  containerColumn.appendChild(containerAddClass(valuecontainerColumn, "column"));
+  containerRow.appendChild(containerAddClass(valuecontainerRow, "row"));
+}
+
+const forrEachArrayValue = function (a, valuecontainerColumn, valuecontainerRow) {
+  a.forEach(function (element) {
+    buildArrayValue(element, valuecontainerColumn)
+    buildArrayValue(element, valuecontainerRow)
+  })
+}
+
+const containerIfValueLine = function (a, containerColumn, containerRow) {
+  buildValueLine(a, containerColumn, "column");
+  buildValueLine(a, containerRow, "row");
+}
+
+const buildValueLine = function (a, container, position) {
+  let valuecontainerRow = document.createElement("div");
+  valuecontainerRow.innerHTML = a;
+  container.appendChild(containerAddClass(valuecontainerRow, position));
+}
+
+const buildValueKey = function (key, container) {
+  let specialistrow = document.createElement("div");
+  specialistrow.innerHTML = key;
+  specialistrow.classList.add("details-notes--italic");
+  container.appendChild(specialistrow);
+}
+
+const buildArrayValue = function (value, container) {
+  let textrow = document.createElement("div");
+  textrow.innerHTML = value;
+  textrow.classList.add("details-notes--margin");
+  container.appendChild(textrow);
+}
+
+const containerAddClass = function (elem, placing) {
```

Comment:
this is okay
I understand that function add class to container
but on realiation i see that it's better to name it like
addPlacementClassesToElement